### ISSUE TITLE
react/require-render-return should not warn if there is no render method (abstract component)

### ISSUE THREAD

**Issue by romainneutron** [yannickcr/eslint-plugin-react#552]
In case I use an "abstract" Component, I used to write them like this:

```js
export default class AbstractComponent extends Component {
    abstractmethod = () => {
        // ...
    }
}
```
then use them
```js
import AbstractComponent from './AbstractComponent';

export default class MyComponent extends AbstractComponent {
   // ...
}
```

With `react/require-render-return`, an error is raised: `Your render method should have return statement`
Should I use another way to use heritage? Is there a missing case in this rule?

**Comment by EvNaverniouk** [comment]
+1. This is required as of 5.0.0. Abstract components were previously being correctly ignored in version 4.x.x but looks like something broke in 5.0.0

**Comment by iongion** [comment]
In my react project I have non-react *ES6* classes that by chance have a `render` method, can the linter detect and shout when encountering only react classes ?

```
class MediaControl extends Clappr.UICorePlugin {
  get name() { return 'media_control'; }
  render() {
    this.rendered = true;
    this.update();
  }
}
```

**Comment by yannickcr** [comment]
@romainneutron and @iongion Can you test with the `v5.1.1` to see if you still have the issue ?

**Comment by iongion** [comment]
@yannickcr I do not get it anymore, awesome!

**Comment by EvNaverniouk** [comment]
@yannickcr Can confirm as well. Seems all good in v5.1.1

**Comment by ljharb** [comment]
Sounds like this is fixed - I'll close this now. Please reopen if there's further issues.
